### PR TITLE
Fix some issues reported by shellcheck

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

**Comment by blueyed** [comment]
Should be squashed when merging.

File: libexec/rbenv-help [hunk 1]
```diff
@@ -112,9 +113,9 @@ print_help() {
   local command="$1"
   local summary usage help
   eval "$(documentation_for "$command")"
-  [ -n "$help" ] || help="$summary"
+  [[ -n "$help" ]] || help="$summary"
```
Comment (mislav) [inline libexec/rbenv-help]: The double `[[` here are not necessary.

File: libexec/rbenv-help [hunk 2]
```diff
@@ -112,9 +113,9 @@ print_help() {
...
 
-  if [ -n "$usage" -o -n "$summary" ]; then
+  if [[ -n "$usage" ]] || [[ -n "$summary" ]]; then
```
Comment (mislav) [inline libexec/rbenv-help]: Ditto `[[`

File: libexec/rbenv-help (hunk 1 above)
Comment (blueyed) [reply libexec/rbenv-help]: Not necessary, but better in fact (`[` is a command, `[[` an instruction), no?

Comment (mislav) [reply libexec/rbenv-help]: In bash, `[` is a builtin (see `help [`), therefore there is no performance saving.

We only use `[[` for pattern matching.

Comment (blueyed) [reply libexec/rbenv-help]: I see, this is different in Zsh then (AFAIK).

Amended / squashed / rebased.

**Pull request merged**